examples/lie_labs.py

```diff
@@ -0,0 +1,121 @@
+import torch
+
+import theseus.labs.lie as lie
+import theseus.labs.lie.functional as lieF
+
+batch_size = 5
+
+# ### Lie Tensor creation functions
+g1 = lie.rand(batch_size, lie.SE3, requires_grad=True)
```

Comment:
Can `batch_size` be an optional argument that defaults to `1`?
I imagine there would be a lot of non-training usecases where we just want a random lie variable.

Reply:
In fact, our plan for the library is to not have a required batch dimension, and support everything from no batch dimension to arbitrary number of batch dimensions. @fantaosha is working on this feature at the moment. In this case, the API for `rand` would be closer to `torch.rand` (e.g., you can do `lie.rand(lie.SE3)`, `lie.rand(4, 5, 6, lie.SE3)`, and so on.

Reply:
Makes sense! However as torch has a default dtype and lie does not (because there's no sensible default), maybe the required parameter should come first to be less confusing? e.g. `lie.rand(lie.SE3, 4, 5, 6)`

Would actually be even cleaner to be able to do
```
lie.SE3.rand()
lie.SE3.rand(4, 5, 6)
```
similar to some of the Sophus initialization methods.

Reply:
The funny thing is that our first version had the `ltype`-first convention, and changed it after some discussion to be more similar to `torch`. But I actually like `lie.SE3.rand` suggestion, this seems cleaner to me.

Reply:
Added in 530172f5005a92b0eb52446092cd3f8620c673d9.